src/compiler/checker.ts

```diff
@@ -1032,6 +1033,18 @@ namespace ts {
             }
         }
 
+        function checkAndReportErrorForUsingTypeAsNamespace(errorLocation: Node, name: string, meaning: SymbolFlags): boolean {
+            if (meaning === SymbolFlags.Namespace) {
+                const symbol = resolveSymbol(resolveName(errorLocation, name, SymbolFlags.Type & ~SymbolFlags.Value, /*nameNotFoundMessage*/undefined, /*nameArg*/ undefined));

```

Comment:
`SymbolFlags.Type & ~SymbolFlags.Value` excludes classes and enums.  Was that intentional? cc @DanielRosenwasser

Reply:
I guess it should have been `& ~SymbolFlags.Namespace` instead

Reply:
I could be missing something, but I don't think `SymbolFlags.Type` includes `SymbolFlags.Namespace`.